sdk-commons/src/main/java/org/openmrs/maven/plugins/utility/PropertiesUtils.java

```diff
@@ -114,4 +138,117 @@ public static void loadPropertiesFromInputStream(InputStream in, Properties prop
+		try {
+			HttpClient httpClient = HttpClientBuilder.create().build();
+			HttpGet request = new HttpGet(url);
+			HttpResponse response = httpClient.execute(request);
+			HttpEntity entity = response.getEntity();
+			try (InputStream inputStream = entity.getContent()) {
+				String jsonData = IOUtils.toString(inputStream, StandardCharsets.UTF_8);
+				ObjectMapper objectMapper = new ObjectMapper();
+				JsonNode jsonNode = objectMapper.readTree(jsonData);
+				JsonNode frontendModules = jsonNode.get("frontendModules");
+				Iterator<Map.Entry<String, JsonNode>> modulesIterator = frontendModules.fields();
+				while (modulesIterator.hasNext()) {
+					Map.Entry<String, JsonNode> moduleEntry = modulesIterator.next();
+					String moduleName = moduleEntry.getKey();
+					String moduleVersion = moduleEntry.getValue().asText();
+					properties.setProperty("spa.frontendModules." + moduleName, moduleVersion);
+				}
+			}
+			catch (IOException e) {
+				log.error(e.getMessage(), e);
+			}
+		}
+		catch (IOException e) {
+			log.error(e.getMessage(), e);
+		}
+		return properties;
+	}
+
+	/**
+	 * Generates configuration properties based on the specified Artifact.
+	 *
+	 * @param artifact The Artifact object containing information about the configuration.
+	 * @return A Properties object containing the generated configuration properties.
+	 */
+	public static Properties getConfigurationProperties(Artifact artifact) {
+		Properties properties = new Properties();
+		properties.setProperty("config." + artifact.getArtifactId(), artifact.getVersion());
+		return properties;
+	}
+
+	/**
+	 * Retrieves backend properties from the specified ZIP file.
+	 *
+	 * @param file The ZIP file to read the backend properties from.
+	 * @return A Properties object containing the retrieved backend properties.
+	 * @throws MojoExecutionException If an error occurs while reading the ZIP file.
+	 */
+	public static Properties getBackendProperties(File file) throws MojoExecutionException {
+		Properties properties = new Properties();
+		try (ZipFile zipFile = new ZipFile(file)) {
+			Enumeration<? extends ZipEntry> entries = zipFile.entries();
+			while (entries.hasMoreElements()) {
+				ZipEntry zipEntry = entries.nextElement();
+				if ("distro.properties".equals(zipEntry.getName())) {
+					properties.load(zipFile.getInputStream(zipEntry));
+				}
+			}
+		}
+		catch (IOException e) {
+			throw new MojoExecutionException("Could not read \"" + file.getAbsolutePath() + "\" " + e.getMessage(), e);
+		}
+		finally {
+			file.delete();
+		}
+		return properties;
+	}
+
+	/**
+	 * Retrieves the latest module property from the specified URL by parsing an XML document.
+	 *
+	 * @param url The URL to retrieve the XML document from.
+	 * @return A Properties object containing the retrieved module property.
+	 * @throws RuntimeException If an error occurs while parsing the XML document.
+	 */
+	public static Properties getLatestModuleProperty(String url) {
```

Comment:
What does latest module property mean?

Reply:
The latest version of the module. I think the wording is not quite right. I'll change it.